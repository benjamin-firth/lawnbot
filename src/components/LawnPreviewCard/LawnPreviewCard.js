import React from 'react';
import styled from 'styled-components';
import ReactCompareImage from 'react-compare-image';

const LawnPreviewCard = ({ beforePic, afterPic, services, location}) => {
  const allServices = services.map(service => <li key={Date.now() * Math.random()} >{service}</li>);

  return (
    <StyledPreviewCard>
      <StyledTop>Yard in: {location}</StyledTop>
      <ReactCompareImage leftImage={beforePic} rightImage={afterPic} />
      <StyledTitle>Services Provided:</StyledTitle>
      <ul>
        {allServices}
      </ul>
    </StyledPreviewCard>
  );
};

const StyledPreviewCard = styled.article`
  background-color: #fff;
  border-radius: 5px;
	box-shadow: 0 10px 6px -6px #777;
  display: flex;
  flex-direction: column;
  margin-top: 20px;
  padding: 5px 25px 15px 25px;
  width: 550px;
`;

const StyledTop = styled.p`
  font-weight: bold;
`;

const StyledTitle = styled.p`
  font-weight: bold;
  margin-bottom: 0px;
`;

export default LawnPreviewCard;