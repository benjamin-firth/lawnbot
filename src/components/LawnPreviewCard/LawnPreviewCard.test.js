import React from 'react';
import { shallow } from 'enzyme';
import LawnPreviewCard from './LawnPreviewCard';

describe('LawnPreviewCard', () => {
  let wrapper;
  const beforePic = 'testUrl';
  const afterPic = 'testUrl';
  const location = 'testLocation';
  const services = ['test service one', 'test two'];

  beforeEach(() => {
    wrapper = shallow(
      <LawnPreviewCard 
        beforePic={beforePic} 
        afterPic={afterPic} 
        location={location} 
        services={services} 
      />
    )
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});