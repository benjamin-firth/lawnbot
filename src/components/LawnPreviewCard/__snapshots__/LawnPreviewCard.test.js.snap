// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LawnPreviewCard should match the snapshot 1`] = `
<styled.article>
  <s
    aspectRatio="taller"
    handle={null}
    handleSize={40}
    hover={false}
    leftImage="testUrl"
    leftImageAlt=""
    leftImageCss={Object {}}
    leftImageLabel={null}
    onSliderPositionChange={[Function]}
    rightImage="testUrl"
    rightImageAlt=""
    rightImageCss={Object {}}
    rightImageLabel={null}
    skeleton={null}
    sliderLineColor="#ffffff"
    sliderLineWidth={2}
    sliderPositionPercentage={0.5}
    vertical={false}
  />
  <p>
    Services Provided:
  </p>
  <ul>
    <li
      key="1434600299396.3542"
    >
      test service one
    </li>
    <li
      key="648168058389.9491"
    >
      test two
    </li>
  </ul>
</styled.article>
`;
